standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -3647,9 +3647,12 @@ long generateCompactionQueueId(Statement stmt) throws SQLException, MetaExceptio
             + "no record found in next_compaction_queue_id");
       }
       long id = rs.getLong(1);
-      s = "UPDATE \"NEXT_COMPACTION_QUEUE_ID\" SET \"NCQ_NEXT\" = " + (id + 1);
+      s = "UPDATE \"NEXT_COMPACTION_QUEUE_ID\" SET \"NCQ_NEXT\" = " + (id + 1) + " WHERE \"NCQ_NEXT\" = " + id;
       LOG.debug("Going to execute update <{}>", s);
-      stmt.executeUpdate(s);
+      if (stmt.executeUpdate(s) != 1) {
```

Comment:
this looks like a hack in production code while we're targeting test fixes here, which is a bit concerning
this method is already very confusing: it looks like cannot guarantee synchronization regarding NCQ_NEXT (which is okay if we have more instances running this code, right?), as we're selecting it, doing something here in java code (increment), then trying to write it back, isn't there a way to use AUTOINCREMENTed value across the supported RDBMS types?
I mean, this recursive call is not something I really like, we should find a better synchronization pattern, I think a compaction queue id increment is not something that happens 100 times per second, so even a RDBMS-level lock is fine on this table I guess (I don't know this area though)

Reply:
This is a short term quick and dirty fix, as the id generation for compaction requests is broken. The old approach did not guarantee a unique id at all, and since Initiator processes the candidates in a parallel way, this could be a production issue too. Actually I have seen this duplicate key error in production logs before. The long term solution is to modify the id to auto increment field. It is already done this way for TXNS table. However, this change will be bigger, therefore I created a separate task for it: https://issues.apache.org/jira/browse/HIVE-27121

Reply:
okay, we can live with a follow-up task, but in this case, I would like to see further minor improvements here:
- HIVE-27121 <- refer to the follow-up task for better reference in a code comment
- use LOG.info at least, this is an unhappy codepath, debug level is not enough
- show the current id in the log message for better reference
-